crypto/property/property.c

```diff
@@ -480,7 +484,8 @@ int ossl_method_store_cache_set(OSSL_METHOD_STORE *store, int nid,
 
     if (method == NULL) {
         elem.query = prop_query;
-        lh_QUERY_delete(alg->cache, &elem);
+        if (lh_QUERY_delete(alg->cache, &elem) != NULL)
+            store->nelem--;
```

Comment:
Doesn't this belong to 9c93e33e8f8889789a249f95cf5259aba9bd03c0?

Reply:
Not really.  It's just keeping the count more accurate.
The count isn't the trigger for the recursive activity -- the `need_flush` flag is.

Reply:
I wasn't too concerned about keeping the count exact when I wrote this.  It would only ever be high and that's not a problem.